### DISCUSSION TITLE
Github recognizing .rsx files as .r files

### DISCUSSION THREAD

**Discussion by TwigYT** [community/community#126604] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I'm looking at the [dioxus repository](https://github.com/DioxusLabs/dioxus), And in the code language breakdown underneath the deployments, it says there is 2.2% R in the repository. But when you search by the R language it recognizes .rsx files as .r files.
Here is an image explaining:
![image](https://github.com/community/community/assets/146784902/f73d0ffc-2bf6-41f5-9784-a915934a0139)

**Comment by look** [comment, marked as answer]
Hi @THG2009, this comes from the [Linguist](https://github.com/github-linguist) language data. Currently, [`.rsx` is associated unambiguously with the R programming language](https://github.com/github-linguist/linguist/blob/e2012cd67867e6287ae5ee96c8fe0b92f9a88efa/lib/linguist/languages.yml#L5682-L5691) which is why the Dioxus RSX files are detected as R.

I would recommend proposing a change to Linguist to either add `.rsx` as a possible suffix for Rust, or perhaps better, a new programming language in the way [TSX](https://github.com/github-linguist/linguist/blob/e2012cd67867e6287ae5ee96c8fe0b92f9a88efa/lib/linguist/languages.yml#L7020-L7030) is.
